docs/connector-development/connector-builder-ui/tutorial.md

```diff
@@ -0,0 +1,202 @@
+
+The warning icon disappears indicating that the declared schema of your stream matches the test data.
+
+## Making the stream configurable
+
+The exchange rate API supports configuring a different base currency via request parameter - let's make this part of the user inputs that can be controlled by the user of the connector when configuring a source, similar to the API key.
+
+To do so, follow these steps:
+* Scroll down to the "Request parameters" section and add a new request parameter
+* Set the key to `base`
+* For the value, click the user icon in the input and select "New user input"
+* Set the name to "Base"
+
+Now your connector has a second configuration input. To test it, click the "Testing values" button again, set the "Base" to `USD`. Then, click the "Test" button again to issue a new test read.
+
+The record should update to use USD as the base currency:
+```
+{
+    "base": "USD",
+    "date": "2023-04-13",
+    "rates": {
+      "AED": 3.6723,
+      "AFN": 86.286329,
+      "ALL": 102.489617,
+      "AMD": 391.984204,
+      // ...
+    }
+}
+```
+
+## Adding incremental reads
+
+We now have a working implementation of a connector reading the latest exchange rates for a given currency.
+In this section, we'll update the source to read historical data instead of only reading the latest exchange rates.
+
+According to the API documentation, we can read the exchange rate for a specific date range by querying the `"/exchangerates_data/{date}"` endpoint instead of `"/exchangerates_data/latest"`.
+
+To configure your connector to request every day individually, follow these steps:
+* Enable "Incremental sync" for the Rates stream
+* Set the "Cursor field" to `date` - this is the property in our records to check what date got synced last
+* Set the "Datetime format" to `%Y-%m-%d` to match the format of the date in the record returned from the API
+* Set the "Cursor granularity" to `P1D` to tell the connector the API only supports daily increments
+* Make the start time configurable by clicking the user icon in the "Start datetime" input and creating a new user input called `Start date`
+* Set the "End datetime" to `{{ today_utc() }}` to always sync up to the current day
+* In a lot of cases the start and end date are injected into the request body or request parameters. However in the case of the exchange rate API it needs to be added to the path of the request, so disable the "Inject start/end time into outgoing HTTP request" options
+* On top of the form, change the "Path URL" input to `/exchangerates_data/{{ stream_slice.start_time }}` to inject the date to fetch data for into the path of the request
+* Open the "Advanced" section and set "Step" to `P1D` to configure the connector to do one separate request per day
+* Set a start date (like `2023-03-03`) in the "Testing values" menu
+* Hit the "Test" button to trigger a new test read
```

Comment:
I'm getting an error

![Screen Shot 2023-04-14 at 11 01 08 AM](https://user-images.githubusercontent.com/1451943/232122224-6c669506-688c-4883-8861-76ae0e66b751.png)

Reply:
Not great, but I was able to fix by formatting the end datetime

![Screen Shot 2023-04-14 at 11 04 24 AM](https://user-images.githubusercontent.com/1451943/232122843-e6dca4f6-da94-4007-a8f7-b2745d25310b.png)

Reply:
I didn't hit any error during this step of the tutorial... not sure why it would be different for us (I'm using CDK version 0.34.1 if that helps). Here is my manifest at this point:
[exchange_rates.yaml.txt](https://github.com/airbytehq/airbyte/files/11237884/exchange_rates.yaml.txt)

Reply:
You didn't set `today_utc` for the end date @girarda - however, I don't think it matters as we are going to remove this before the release to avoid this kind of error.

Reply:
Note: We will need to adjust the tutorial a bit once that's done, but I think we don't need to block on anything